File: main/src/main/java/com/sedmelluq/discord/lavaplayer/container/ogg/OggMetadata.java
package com.sedmelluq.discord.lavaplayer.container.ogg;

import com.sedmelluq.discord.lavaplayer.track.info.AudioTrackInfoProvider;

import java.util.Collections;
import java.util.Map;

/**
 * Audio track info provider based on OGG metadata map.
 */
public class OggMetadata implements AudioTrackInfoProvider {
  public static final OggMetadata EMPTY = new OggMetadata(Collections.emptyMap());

  private static final String TITLE_FIELD = "TITLE";
  private static final String ARTIST_FIELD = "ARTIST";

  private final Map<String, String> tags;

  /**
   * @param tags Map of OGG metadata with OGG-specific keys.
   */
  public OggMetadata(Map<String, String> tags) {
    this.tags = tags;
  }

  @Override
  public String getTitle() {
    return tags.get(TITLE_FIELD);
  }

  @Override
  public String getAuthor() {
    return tags.get(ARTIST_FIELD);
  }

  @Override
  public Long getLength() {
    return null;
  }

  @Override
  public String getIdentifier() {
    return null;
  }

  @Override
  public String getUri() {
    return null;
  }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/container/ogg/OggMetadata.kt
package com.sedmelluq.discord.lavaplayer.container.ogg

import com.sedmelluq.discord.lavaplayer.track.info.AudioTrackInfoProvider
import com.sedmelluq.discord.lavaplayer.container.ogg.OggMetadata
import com.sedmelluq.discord.lavaplayer.tools.Units

/**
 * Audio track info provider based on OGG metadata map.
 *
 * @param tags Map of OGG metadata with OGG-specific keys.
 */
class OggMetadata(private val tags: Map<String, String>, override val length: Long?) : AudioTrackInfoProvider {
    companion object {
        @JvmField
        val EMPTY = OggMetadata(emptyMap(), Units.DURATION_MS_UNKNOWN)
        private const val TITLE_FIELD = "TITLE"
        private const val ARTIST_FIELD = "ARTIST"
    }

    override val title: String?
        get() = tags[TITLE_FIELD]

    override val author: String?
        get() = tags[ARTIST_FIELD]

    override val identifier: String?
        get() = null

    override val uri: String?
        get() = null

    override val artworkUrl: String?
        get() = null
}
